### ISSUE TITLE
V540TND v0.9.1-rc5 relatively long boot times

### ISSUE THREAD

**Issue by philipandag** [Dasharo/dasharo-issues#1049]
### Component

Dasharo firmware

### Device

NovaCustom V54 14th Gen

### Dasharo version

v0.9.1-rc5

### Dasharo Tools Suite version

develop

### Test case ID

CBMEM002, CBMEM003

### Brief summary

Very long boot times, exceeding the limits given in platform configs

### How reproducible

I have only run the suite once.

### How to reproduce

Run CBMEM tests via osfv

### Expected behavior

Average boot times should be < 10s.

### Actual behavior

 In this case most are ~7-8s but because of a large spike on the first boot time the average warmboot time exceeded 10s.

### Screenshots

[dasharo-performance_log.zip](https://github.com/user-attachments/files/16989613/dasharo-performance_log.zip)

### Additional context

I have seen large boot time spikes on the first measurement before. Maybe it's an error in measurements?

**Comment by mkopec** [comment]
15s in coreboot is a long time, especially on a warmboot with memory training already cached, this is probably a bug in firmware

**Comment by SebastianCzapla** [comment]
With rc6 I got longer boot time once, when picking up laptop for testing. I snapped cbmem -1 log. Also, systemd-analyze showed 16s in firmware.
Despite that, running suite does passes with rc6, and manual retesting gives no failing state. I speculate that with rc6 this is not fixed, but happens rarely.

cbmem no args was made after the suite passed.

[cbmem-noargs-log-longboot.txt](https://github.com/user-attachments/files/17281253/cbmem-noargs-log-longboot.txt)
[cbmem-1-log-longboot.txt](https://github.com/user-attachments/files/17281254/cbmem-1-log-longboot.txt)
[boot-time-measure.robot_log.zip](https://github.com/user-attachments/files/17281284/boot-time-measure.robot_log.zip)

**Comment by miczyg1** [comment]
Also the test log only check for the coreboot boot time: `[1.956179, 3.136912, 2.326149, 3.088181, 2.413734] `
These logs do not have anything suspicious in it. if we want to detect such issues, we should use `systemd-analyze` given that firmware performance measurement is enabled in the builds

**Comment by miczyg1** [comment]
> In this case most are ~7-8s but because of a large spike on the first boot time the average warmboot time exceeded 10s.

First boot is always longer (if the memory was trained earlier), because EDK2 has to format and initialize variable storage.

**Comment by miczyg1** [comment]
From the lgos attached in the issue description:

1st boot: `951	35408040	34262350	returning from FspMemoryInit` 34s for RAM training, expected.
All subsequent boots have 8.x seconds: ` [42.252427, 8.43205, 8.442174, 8.38789, 8.410944] ]`

The first boot measuement is what screws up the max and average boot time.
Suggestion:

1. Do 6 reps of the measurements and always ignore first iterations.
2. Try to interpret the results first? It should be an obvious thing that first boot is always long due to RAM training. We repeat it over and over again multiple times, mentioned in documentation for many platforms...

tl;dr

Not an FW issue, improve the test.
